src/svc/telemetry/prometheus.rs

```diff
@@ -125,10 +175,7 @@ pub fn convert_metrics_to_prometheus(
 }
 
 /// assign worker_id and cluster_id as labels
-fn apply_labels(
-    aggregated_metrics: AggregatedMetrics,
-    aggregate_backend_metrics: bool,
-) -> Vec<LabeledMetric> {
+fn apply_labels(aggregated_metrics: AggregatedMetrics) -> Vec<LabeledMetric> {
     let mut labeled_metrics = Vec::new();
```

Comment:
Iterator, flat_map, collect

Reply:
we will optimize the code if we find this to be a bottleneck in production